client.go

```diff
@@ -0,0 +1,104 @@
+package gotruevault
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+const (
+	ContentTypeApplicationJSON = "application/json"
+)
+
+var (
+	// ErrUnauthorized ...
+	ErrUnauthorized = errors.New("error: authorization failed")
+
+	// ErrServerError ...
+	ErrServerError = errors.New("error: server error")
+
+	// ErrBadRequest ...
+	ErrBadRequest = errors.New("error: bad request")
+)
+
+//go:generate mockery --name URLBuilder
+type URLBuilder interface {
+	SearchDocumentURL(vaultID string) string
+}
+
+// defaultURLBuilder  ...
+type defaultURLBuilder struct{}
+
+// SearchDocumentURL ...
+func (t *defaultURLBuilder) SearchDocumentURL(vaultID string) string {
```

Comment:
Why not just do this in the function that needs it?

Reply:
Because i want to dependency inject the URL builder when doing a test. See this [line](https://github.com/FirstVisit/go-truevault/pull/1/files#diff-51ee3eebe09cbb7c5561e4078f0d864fa0325edd2724b05dd48758211f56ea1dR49)